### PR TITLE
Feature 1134 enable informants officers create edit pathogen tests

### PR DESCRIPTION
**Pull request by opatajoshua**
#1134

### CODE REVIEW

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/sample/PathogenTestDtoHelper.java [hunk 1]
```diff
@@ -63,6 +69,12 @@ protected void fillInnerFromDto(PathogenTest target, PathogenTestDto source) {
         target.setTestType(source.getTestType());
         target.setTestedDisease(source.getTestedDisease());
         target.setTestedDiseaseDetails(source.getTestedDiseaseDetails());
+
+        //newly added
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/backend/sample/PathogenTestDtoHelper.java]: Please only add comments that provide relevant information to understand the code or what it's supposed to do

File: sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditActivity.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Override
+    protected BaseEditFragment buildEditFragment(PageMenuItem menuItem, PathogenTest activityRootData) {
+        return PathogenTestEditFragment.newInstance(activityRootData);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        super.onCreateOptionsMenu(menu);
+        getSaveMenu().setTitle(R.string.action_save_contact);
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditActivity.java]: If you copy existing classes to write new ones with similar functionality (which is completely valid), please make sure to rename everything that was semantically associated with the old class (e.g. in this case, it should be R.string.action_save_pathogen_test)

File: sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditFragment.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    private List<Item> testResultList;
+
+    // Instance methods
+
+    public static PathogenTestEditFragment newInstance(PathogenTest activityRootData) {
+        return newInstance(PathogenTestEditFragment.class, null, activityRootData);
+    }
+
+    private void setUpControlListeners(FragmentPathogenTestEditLayoutBinding contentBinding) {
+
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditFragment.java]: Remove empty methods that aren't used

File: sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditFragment.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        testResultList = DataUtils.getEnumItems(PathogenTestResultType.class, true);
+        labList = DatabaseHelper.getFacilityDao().getLaboratories(true);
+    }
+
+    @Override
+    public void onLayoutBinding(FragmentPathogenTestEditLayoutBinding contentBinding) {
+        setUpControlListeners(contentBinding);
+
+        contentBinding.setData(record);
+//        contentBinding.setCaze(sourceCase);
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditFragment.java]: Please remove commented code that is no longer used, especially if it's a leftover from copying a class

**Comment by MateStrysewskeSym** [comment]
@opatajoshua I've gone through both of your pull requests and added comments to the changed files (please note that I didn't check everything, but only added comments as examples). Please have a look at them and adjust the code accordingly. 

@faatihi Since you're already experienced in working with SORMAS, it's probably a good idea to review the overhauled code together with Joshua.

**Comment by opatajoshua** [comment]
Done with the comment issues and formatting

**Comment by MateStrysewskeSym** [comment]
@opatajoshua I think something went wrong during the merging, there are merging artifacts in the code you pushed.

**Comment by MartinWahnschaffeSymeda** [comment]
@opatajoshua I have reviewed the pull request again and would like to commit a few changes I have made. Can you give me access to edit the pull request, please?
See https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request merged**